configure.ac

```diff
@@ -114,7 +114,7 @@ AC_ARG_ENABLE([tests],
 AS_IF([test "${enable_tests}" != "no" -a "${enable_neuron}" = "yes"],
       [AC_MSG_WARN([Disabling tests due to Neuron configuration.])
        enable_tests=no],
-      [test "${enable_tests}" = "yes" -a "${found_mpi}" = "no"],
+      [test "${enable_tests}" != "no" -a "${found_mpi}" = "no"],
```

Comment:
I don't think this is right, and it certainly isn't a fix (change yes, fix no).  We don't want to require MPI for normal users to build, and there's no real value in the tests for end users, so why force everyone to have MPI?

Reply:
I'm not following. This line makes sure that if tests are enabled, MPI is found. There was a bug, in that `test "${enable_tests}" = "yes"` is not the correct way to check if tests are enabled.

Whether tests are enabled by default is a different question.

Reply:
`test "${enable_tests}" = "yes"` absolutely is the right way to check if the customer explicitly enabled the tests.  Why do you think it isn't?

Reply:
Ok, sorry, I understand what you're saying now.

> `test "${enable_tests}" = "yes"` absolutely is the right way to check if the customer explicitly enabled the tests.

Agree. The problem is that we are currently enabling tests by default, without the customer explicitly enabling them.

If the customer does not explicitly enable tests, `test "${enable_tests}" = "yes"` will evaluate to false, so we won't check for MPI. But we will try to build the tests anyway and fail during build-time.

So, we can either make this change, or change to tests-disabled-by-default. But the code today will pass configure (implicitly enabling tests) and then fail during build if MPI isn't found, which is wrong.